Mathlib/Data/Set/Function.lean

```diff
@@ -624,6 +624,45 @@ lemma bijOn_id (s : Set α) : BijOn id s s := ⟨s.mapsTo_id, s.injOn_id, s.surj
 theorem BijOn.comp (hg : BijOn g t p) (hf : BijOn f s t) : BijOn (g ∘ f) s p :=
   BijOn.mk (hg.mapsTo.comp hf.mapsTo) (hg.injOn.comp hf.injOn hf.mapsTo) (hg.surjOn.comp hf.surjOn)
 
+/-- If `f : α → β` and `g : β → γ` and if `f` is injective on `s`, then `f ∘ g` is a bijection
+on `s` iff  `g` is a bijection on `f '' s`. -/
+theorem bijOn_comp_iff_bijOn_image {α β γ : Type*} {f : α → β} {g : β → γ} {s : Set α}
+    (hf : InjOn f s) {t : Set γ} : BijOn (g ∘ f) s t ↔ BijOn g (f '' s) t := by
+  constructor
+  · rintro ⟨h₁, h₂, h₃⟩
+    refine ⟨mapsTo_image_iff.mpr h₁, InjOn.image_of_comp h₂, fun x hx ↦ ?_⟩
+    obtain ⟨y, hy, rfl⟩ := h₃ hx
+    exact ⟨f y, ⟨y, hy, rfl⟩, rfl⟩
+  · rintro ⟨h₁, h₂, h₃⟩
+    exact ⟨mapsTo_image_iff.mp h₁, InjOn.comp h₂ hf <| mapsTo_image f s,
+      SurjOn.comp h₃ <| surjOn_image _ _⟩
+
+/--
+If we have a commutative square
+
+α --f--> β
+|        |
+p₁       p₂
+|        |
+\/       \/
+γ --g--> δ
```

Comment:
I think that you should enclose the diagram into triple backticks, otherwise it'll be completely messed up in the HTML docs.